### PR TITLE
Create boostrap version of gce-flaky

### PR DESCRIPTION
**Pull request by fejta**
1) Add line to `jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml`
2) Copy `ci-kubernetes-e2e-gce-etcd3.sh` to `ci-kubernetes-e2d-gce-flaky.sh`
3) Replace job-env with the info from `jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gce.yaml`

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1020)

### CODE REVIEW

**Comment by ixdy** [comment]
One thing we need to think about: [find_version](https://github.com/kubernetes/test-infra/blob/master/jenkins/bootstrap.py#L335-L358) won't work correctly for CI right now.

**Comment by fejta** [comment]
Seems to work just fine? https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-etcd3/1015/build-log.txt

**Comment by spxtr** [comment]
I would be surprised if that version is always correct. The actual version under test is what is in the tar that we download, not what we checkout.

**Comment by ixdy** [comment]
I think that's only happening to work by coincidence.

boostrap.py is checking out the kubernetes repo from GitHub and then using whatever version it happened to check out to fill in `started.json`, but then e2e-runner.sh checks out the latest built binaries. In the example you linked, they happen to be the same version, but it's not hard to imagine a situation where this is not true.

**Comment by ixdy** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-etcd3/1014?log shows a version mismatch.

We report the tested version as v1.6.0-alpha.0.323+3bd87044897a31, but the `kubectl version` output doesn't match:

```
W1109 13:27:02.774] 2016/11/09 13:27:02 e2e.go:722: Running: kubectl version
I1109 13:27:02.892] Client Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.321+8b5264e095251a", GitCommit:"8b5264e095251aa4ffa546390e2ef785061dc5e8", GitTreeState:"clean", BuildDate:"2016-11-09T20:26:42Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}
I1109 13:27:02.937] Server Version: version.Info{Major:"1", Minor:"6+", GitVersion:"v1.6.0-alpha.0.321+8b5264e095251a", GitCommit:"8b5264e095251aa4ffa546390e2ef785061dc5e8", GitTreeState:"clean", BuildDate:"2016-11-09T20:23:21Z", GoVersion:"go1.7.1", Compiler:"gc", Platform:"linux/amd64"}

```

**Comment by fejta** [comment]
Makes sense thanks for the explanation!

File: jobs/ci-kubernetes-e2e-gce-flaky.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+set -o xtrace
+
+readonly testinfra="$(dirname "${0}")/.."
+
+### provider-env
+export KUBERNETES_PROVIDER="gce"
+export E2E_MIN_STARTUP_PODS="8"
+export KUBE_GCE_ZONE="us-central1-f"
+export FAIL_ON_GCP_RESOURCE_LEAK="true"
+export CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS="1"
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-flaky.sh]: later on will we be able to reuse these envs for other gce jobs?

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -893,6 +893,8 @@ test_groups:
   gcs_prefix: kubernetes-jenkins/logs/kubernetes-update-jenkins-jobs
 - name: ci-kubernetes-e2e-gce-etcd3
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-etcd3
+- name: ci-kubernetes-e2e-gce-flaky
+  gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-flaky
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: add the tab to dashboard :-)

File: jobs/ci-kubernetes-e2e-gce-flaky.sh (hunk 1 above)
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-flaky.sh]: -- I mean, maybe, have an equivalent global file here?

Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-flaky.sh]: Yes that is the next step after moving over everything (reimplement https://github.com/kubernetes/test-infra/pull/435)

File: testgrid/config/config.yaml (hunk 2 above)
Comment (fejta) [inline testgrid/config/config.yaml]: Done

**Comment by krzyzacy** [comment]
Is our strategy is convert and let both running until we verify bootstrap-ci is working, and then get rid of the original job?

If so shall we consider lower down the frequency for the bootstrap jobs for now? Since if I massively convert them then it will be flooded.

**Comment by krzyzacy** [comment]
/lgtm

Feel free to merge and I'd like to see it runs fine first.

File: jobs/ci-kubernetes-e2e-gce-flaky.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+# GKE variables
+export CLUSTER_NAME="${E2E_NAME}"
+export KUBE_GKE_NETWORK="${E2E_NAME}"
+
+# Get golang into our PATH so we can run e2e.go
+export PATH="${PATH}:/usr/local/go/bin"
+
+### Runner
+readonly runner="${testinfra}/jenkins/dockerized-e2e-runner.sh"
+timeout -k 15m 50m "${runner}" && rc=$? || rc=$?
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-flaky.sh]: em, just a question, how this timeout is calculated, since original it was `timeout: 180`

**Comment by spxtr** [comment]
LGTM

File: jobs/ci-kubernetes-e2e-gce-flaky.sh (hunk 3 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-flaky.sh]: Good catch!

**Comment by fejta** [comment]
I believe the second commit will fix #1032

**Pull request merged**